File: 백준/Silver/1912. 연속합/연속합.py
import sys
fi = sys.stdin.readline
                
def solution():
    n = int(fi().rstrip())
    arr = list(map(int, fi().split()))
    # print(*arr, end=" ")
    d = [arr[0]]
    
    for i in range(1, n):
        # 더 큰 수가 될때만 더하기
        d.append(max(d[-1]+arr[i], arr[i]))
        
    print(max(d))
    
if __name__ == '__main__':
    solution()

File: 백준/Silver/1912. 연속합/연속합.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        
        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n];
        
        StringTokenizer st = new StringTokenizer(br.readLine());
        for(int i=0; i<n;i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }
        
        int[] pfx = new int[n+1];
        Arrays.fill(pfx, Integer.MIN_VALUE+1001);

        for(int i=0;i<n;i++) {
            pfx[i+1] = Math.max(pfx[i] + arr[i], arr[i]);
        }
        
        bw.write(Arrays.stream(pfx).max().orElseThrow() + "");
        bw.flush();
        bw.close();
    }
}
